pkg/kubernetes/cluster.go

```diff
@@ -74,20 +78,10 @@ func (c *Cluster) DeleteResource(ctx context.Context, name, namespace, kind stri
 	// get REST interface
 	dr := dynClient.Resource(mapping.Resource).Namespace(namespace)
 
-	// check if the resource is there
-	res, err := dr.Get(context.TODO(), name, metav1.GetOptions{})
-	if res == nil {
-		// already gone, no delete necessary
-		return nil
-	}
-	if err != nil {
-		return fmt.Errorf("failed fetching the resource: %w", err)
-	}
-
 	err = dr.Delete(ctx, name, metav1.DeleteOptions{})
-	if err != nil {
+	if !k8serr.IsNotFound(err) {
 		return err
 	}
-
+	c.logger.Print("resource not found, no need to delete")
```

Comment:
Shouldn't you also mention which resource is this about? I see you mention it in a log before, but assuming 2+ delete operations at the same time will be difficult to find out which one this line is referring to.

Reply:
Right, let me fix it.

Reply:
Ah, no. I just looked at it and it should not be necessary when deleting of each resource starts with 
`want to delete resource %s of kind %s in %s namespace"`

So this message ("resource not found") should occur right after the previous one specifying the resource name and kind.